torchx/lib/torchx/backend.ex

```diff
@@ -921,6 +921,29 @@ defmodule Torchx.Backend do
     |> to_nx(out)
   end
 
+  def dot(
+        %T{type: out_type} = out,
+        %T{type: left_type, data: %TB{ref: left_ref}},
+        left_axes,
+        left_batched_axes,
+        %T{type: right_type, data: %TB{ref: right_ref}},
+        right_axes,
+        right_batched_axes
+      ) do
+    left_axes = Enum.map(left_axes, fn x -> x - Enum.count(left_batched_axes, &(&1 < x)) end)
```

Comment:
@josevalim @jonatanklosko any idea on how to make this more efficient? Basically, I need to "remove" the batching axes from the axis numbering for libtorch

Reply:
You could do manual recursion to avoid multiple traversals but it probably doesn't matter.